scrapy/downloadermiddlewares/retry.py

```diff
@@ -23,12 +23,13 @@
 
 retry_logger = getLogger(__name__)
 
+DEPRECATED_ATTRIBUTE = "EXCEPTIONS_TO_RETRY"
 
-class BackwardsCompatibilityMetaclass(type):
-    @property
-    def EXCEPTIONS_TO_RETRY(cls):
+
+def backwards_compatibility_getattr(self, name):
+    if name == DEPRECATED_ATTRIBUTE:
         warnings.warn(
-            "Attribute RetryMiddleware.EXCEPTIONS_TO_RETRY is deprecated. "
+            f"Attribute RetryMiddleware.{DEPRECATED_ATTRIBUTE} is deprecated. "
             "Use the RETRY_EXCEPTIONS setting instead.",
```

Comment:
I think we should either have both old/new attribute names as constants, or none of them.